### PR TITLE
Add more *_SUPER_MAGIC defines from the latest Linux kernel

### PR DESCRIPTION
**Pull request by asomers**
Over in Nix I have a need for `OVERLAYFS_SUPER_MAGIC`.  Some of these definitions probably apply for Android and uclibc too, but I don't have the sources for those handy.

### CODE REVIEW

File: src/unix/linux_like/linux/gnu/mod.rs [hunk 1]
```diff
@@ -664,32 +664,50 @@ pub const NI_MAXHOST: ::socklen_t = 1025;
 
 pub const ADFS_SUPER_MAGIC: ::c_long = 0x0000adf5;
 pub const AFFS_SUPER_MAGIC: ::c_long = 0x0000adff;
+pub const AFS_SUPER_MAGIC: ::c_long = 0x5346414F;
+pub const AUTOFS_SUPER_MAGIC: ::c_long = 0x0187;
+pub const BINDERFS_SUPER_MAGIC: ::c_long = 0x6c6f6f70;
+pub const BTRFS_SUPER_MAGIC: ::c_long = 0x9123683E;
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/gnu/mod.rs:670]:
```suggestion
pub const BTRFS_SUPER_MAGIC: ::c_long = 0x9123683e;
```

Same here.

File: src/unix/linux_like/linux/gnu/mod.rs [hunk 2]
```diff
@@ -664,32 +664,50 @@ pub const NI_MAXHOST: ::socklen_t = 1025;
...
+pub const CGROUP_SUPER_MAGIC: ::c_long = 0x27e0eb;
 pub const CODA_SUPER_MAGIC: ::c_long = 0x73757245;
 pub const CRAMFS_MAGIC: ::c_long = 0x28cd3d45;
+pub const DEVPTS_SUPER_MAGIC: ::c_long = 0x1cd1;
+pub const ECRYPTFS_SUPER_MAGIC: ::c_long = 0xf15f;
 pub const EFS_SUPER_MAGIC: ::c_long = 0x00414a53;
 pub const EXT2_SUPER_MAGIC: ::c_long = 0x0000ef53;
 pub const EXT3_SUPER_MAGIC: ::c_long = 0x0000ef53;
 pub const EXT4_SUPER_MAGIC: ::c_long = 0x0000ef53;
+pub const F2FS_SUPER_MAGIC: ::c_long = 0xF2F52010;
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/gnu/mod.rs:681]:
```suggestion
pub const F2FS_SUPER_MAGIC: ::c_long = 0xf2f52010;
```

Same here.

File: src/unix/linux_like/linux/gnu/mod.rs [hunk 3]
```diff
@@ -664,32 +664,50 @@ pub const NI_MAXHOST: ::socklen_t = 1025;
...
+pub const FUTEXFS_SUPER_MAGIC: ::c_long = 0xBAD1DEA;
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/gnu/mod.rs:682]:
```suggestion
pub const FUTEXFS_SUPER_MAGIC: ::c_long = 0xbad1dea;
```

glibc defines with uppercase but let's unify here.

File: src/unix/linux_like/linux/gnu/mod.rs (hunk 1 above)
Comment (JohnTitor) [reply src/unix/linux_like/linux/gnu/mod.rs:670]: This one isn't resolved. Also it'd be great if you could squash commits to clean up history.

Comment (asomers) [reply src/unix/linux_like/linux/gnu/mod.rs:670]: What do you mean it isn't resolved?  The CI checks passed.  Do you mean that it couldn't be resolve on your own dev machine?  Are your kernel headers too old?

Comment (JohnTitor) [reply src/unix/linux_like/linux/gnu/mod.rs:670]: Hm? The lowercase style issue as I mentioned is unresolved here.

Comment (asomers) [reply src/unix/linux_like/linux/gnu/mod.rs:670]: Oh, I see what you mean.  I thought you meant that a symbol was unresolved by the compiler.

Comment (asomers) [reply src/unix/linux_like/linux/gnu/mod.rs]: Ok, I squashed all of the commits except the first.  I think the file's history is more understandable when purely stylistic changes like that one are separated from content changes.

Comment (JohnTitor) [reply src/unix/linux_like/linux/gnu/mod.rs]: Ah, sorry for confusing! I'll merge this once CI passes, thanks <3

**Pull request merged**